intro-to-dev-environment/intro-to-computer-memory.md

```diff
@@ -0,0 +1,166 @@
+
+Computers are programmable machines that carry out sequences of instructions and operations. All of the modern computing devices that will run our code will have a memory component. When our code runs, our computers use this memory component to hold onto information like values and variables.
+
+What does it mean when our Activity Monitor says a program takes 100% CPU utilization? What do we mean when we say "memory"? Learning a few things about computer memory can help us be more comfortable debugging.
+
+## Vocabulary and Synonyms
+
+- CPU
+- RAM
+- Stack
+- Heap
+
+## CPU
+
+Computers have a CPU, or a Central Processing Unit. The CPU is a physical bunch of electronic circuitry located in a small place on a motherboard. The CPU's responsibility is to do something very important: execute instructions from software.
+
+We can compare the CPU to a human brain. They're both a physical object specialized in taking input data and _processing_ it.
+
+The CPU is amazing! The CPU itself performs basic math, logic, control, and reading and outputting data. However, with good software (including an operating system), and the help of other computer components, the CPU runs even the most complex programs.
+
+## Physical Memory
+
+The CPU will process many instructions that require data to be stored.
+
+Physical memory devices can hold this data. The CPU will use physical memory by storing data on it and reading data from it.
+
+Memory devices organize the data it stores by having and using **memory addresses**.
+
+Each piece of data on memory will occupy a memory location, which has a memory address.
+
+### Metaphor: Storage Units With Addresses
+
+We can compare a memory device to a [Self-Storage Unit Center](https://en.wikipedia.org/wiki/Self_storage), where each self-storage unit has an address like 100, 101, 102, etc. We can put the data from a `taxes` program into the storage units with the addresses 301, 302, and 303. We can put the data from a `weather` program into the storage units 304, 305, and 306.
+
+When I want to get my data and possessions from my `taxes` program, I'll go to units 301, 302, and 303, and grab whatever is in there. To get data from my `weather` program, I'll go to units 304, 305, and 306, and get what's in those units.
+
+If I need to change my `weather` program data, I'll go to units 304, 305, and 306, and add in my rain data and my rain boots. The next time I _get_ data from 304, 305, and 306, I'll end up grabbing my rain data and rain boots too!
+
+## RAM
+
+RAM, or Random-access memory, is a form of computer memory. RAM sticks are physical memory devices that attach to a motherboard. RAM has two unique traits compared to other forms of computer memory:
+
+1. RAM is fast
+1. RAM holds data temporarily
+
+Ram is fast! The phrase "random-access memory" refers to how RAM is fast; we can say it takes the same amount of time to "randomly access" anything in memory, no matter where it's physically located on the RAM stick.
+
```

Comment:
@anselrognlie 

I like this info! It helps me because I can contrast this concept against RAM, and therefore understand RAM better.

Just out of curiosity, is there a different learning goal you'd want out of this? AKA in a practical sense, is it helpful for an engineer to know this?

(This question could sound judgy, but not trying to be-- Just trying to figure out how much I should desire the instructors to be able to dive into this. Should I give them a note to research this for the benefit of our students etc etc)